File: web/src/lib/category-icons.ts
/**
 * Web-side slug → icon-file bridge for category, cuisine & attribute SVGs.
 *
 * The icons live in web/public/icons/ as Cyrillic-named SVGs, copied from
 * mobile/assets/icons/ by scripts/sync-icons.mjs. The canonical slug↔Cyrillic
 * mapping is backend-only (backend/src/constants/urlSlugs.js — ESM, not
 * web-importable), so this is a MANUAL-SYNC port: if a category, cuisine, or
 * attribute slug changes there, mirror it here AND in sync-icons.mjs.
 *
 * Attribute (amenity) icons were added 2026-06-21 for the vitrine revision,
 * superseding the 2026-05-26 "web keeps lucide-react for amenities" decision —
 * the design handoff uses the brand SVG set across both surfaces.
 *
 * This map resolves icon FILE basenames, not display names. They differ for a
 * couple of entries: category `fast-food` (display «Фаст-фуд») ships as
 * `ФастФуд.svg` (camelCase, no hyphen); attribute `terrace` (display «Летняя
 * терраса») ships as `Терасса.svg`. Display NAMES come from getMetadata() /
 * facets.ts; never derive a label from this map.
 */

/** Category slug → icon file basename (Cyrillic, no extension). 15 entries. */
export const CATEGORY_ICON_BY_SLUG: Record<string, string> = {
  restaurants: 'Ресторан',
  cafes: 'Кафе',
  coffee: 'Кофейня',
  bars: 'Бар',
  pizza: 'Пиццерия',
  bakery: 'Пекарня',
  patisserie: 'Кондитерская',
  'fast-food': 'ФастФуд', // override: display «Фаст-фуд», file «ФастФуд.svg»
  pubs: 'Паб',
  canteens: 'Столовая',
  hookah: 'Кальянная',
  bowling: 'Боулинг',
  karaoke: 'Караоке',
  billiards: 'Бильярд',
  clubs: 'Клуб',
};

/** Cuisine slug → icon file basename (Cyrillic, no extension). 12 entries. */
export const CUISINE_ICON_BY_SLUG: Record<string, string> = {
  belarusian: 'Народная',
  signature: 'Авторская',
  asian: 'Азиатская',
  american: 'Американская',
  vegetarian: 'Вегетарианская',
  japanese: 'Японская',
  georgian: 'Грузинская',
  italian: 'Итальянская',
  mixed: 'Смешанная',
  european: 'Европейская',
  chinese: 'Китайская',
  eastern: 'Восточная',
};

/**
 * Attribute (amenity) slug → icon file basename (Cyrillic, no extension).
 * 9 entries — keys match facets.ts ATTRIBUTE_OPTIONS / establishment-helpers
 * ATTRIBUTE_ORDER exactly. Note `terrace` → «Терасса» (file spelled with one
 * р / two с, even though the display label is «Летняя терраса»).
 */
export const ATTRIBUTE_ICON_BY_SLUG: Record<string, string> = {
  delivery: 'Доставка еды',
  wifi: 'Wifi',
  terrace: 'Терасса',
  parking: 'Парковка',
  live_music: 'Живая музыка',
  kids_zone: 'Детская зона',
  banquet: 'Банкет',
  pets_allowed: 'Животные',
  smoking: 'Курение',
};

/**
 * Resolve a category, cuisine, OR attribute slug to its public icon URL, or
 * null when the slug has no mapped icon (caller may then render a fallback).
 * The three slug namespaces are disjoint, so the merged lookup is unambiguous.
 *
 * The Cyrillic basename is URL-encoded so non-ASCII (and the spaced names like
 * «Доставка еды») resolve predictably regardless of how the request is
 * serialized.
 */
export function iconUrlForSlug(slug: string): string | null {
  const file =
    CATEGORY_ICON_BY_SLUG[slug] ??
    CUISINE_ICON_BY_SLUG[slug] ??
    ATTRIBUTE_ICON_BY_SLUG[slug];
  if (!file) return null;
  return `/icons/${encodeURIComponent(file)}.svg`;
}

File: admin-web/lib/config/category_icons.dart
/// Категория или кухня заведения → имя файла в `assets/icons/`.
///
/// Ключ — то, что реально приходит из API: канон хранит категории и кухни
/// кириллическими названиями (`establishments.categories`,
/// `backend/src/constants/establishmentVocab.js` — 15 категорий, 12 кухонь),
/// а слаги живут отдельно и только для URL (`urlSlugs.js`). Поэтому карта
/// ключуется названием, а не слагом — в отличие от веба, где то же самое
/// решается наоборот (`web/src/lib/category-icons.ts`).
///
/// **Набор иконок здесь неполный.** В `admin-web/assets/icons/` лежат 15
/// файлов — образец из `docs/design_handoff_web_vitrine/assets/icons`, а не
/// весь канон. Полный набор (37 файлов, кириллические имена) — в
/// `mobile/assets/icons/`. Непокрытые категории (Пиццерия, Кальянная,
/// Боулинг и ещё семь) отдают `null`, и вызывающий рисует нейтральный глиф.
/// Врать иконкой «ресторан» там, где боулинг, хуже, чем не показать ничего.
library;

/// Категории, у которых иконка есть. 5 из 15 канонических.
const Map<String, String> _categoryIcons = <String, String>{
  'ресторан': 'restaurant',
  'кофейня': 'coffee',
  'кафе': 'cafe',
  'бар': 'bar',
  'пекарня': 'bakery',
};

/// Кухни, у которых иконка есть. 6 из 12 канонических.
const Map<String, String> _cuisineIcons = <String, String>{
  'народная': 'folk',
  'азиатская': 'asian',
  'японская': 'japanese',
  'грузинская': 'georgian',
  'итальянская': 'italian',
  'европейская': 'european',
};

/// Приведение ключа: регистр и «ё» не должны решать, найдётся иконка или нет.
String _normalize(String value) => value.trim().toLowerCase().replaceAll('ё', 'е');

/// Иконка для карточки заведения: сначала категория, затем кухня.
///
/// Каскад именно такой, потому что категория отвечает на вопрос «что это за
/// место», а кухня лишь уточняет. Возвращает базовое имя файла без пути и
/// расширения либо `null`, если ни одно из названий не покрыто набором.
String? iconNameForEstablishment({
  List<String> categories = const <String>[],
  List<String> cuisines = const <String>[],
}) {
  for (final category in categories) {
    final icon = _categoryIcons[_normalize(category)];
    if (icon != null) return icon;
  }
  for (final cuisine in cuisines) {
    final icon = _cuisineIcons[_normalize(cuisine)];
    if (icon != null) return icon;
  }
  return null;
}

/// Полный путь к SVG для `SvgPicture.asset`, либо `null`.
String? iconAssetForEstablishment({
  List<String> categories = const <String>[],
  List<String> cuisines = const <String>[],
}) {
  final name = iconNameForEstablishment(
    categories: categories,
    cuisines: cuisines,
  );
  return name == null ? null : 'assets/icons/$name.svg';
}
